src/sys/signal.rs

```diff
@@ -410,6 +410,51 @@ pub unsafe fn sigaction(signal: Signal, sigaction: &SigAction) -> Result<SigActi
     Errno::result(res).map(|_| SigAction { sigaction: oldact })
 }
 
+/// Signal management (see [signal(3p)](http://pubs.opengroup.org/onlinepubs/9699919799/functions/signal.html))
+///
+/// Installs `handler` for the given `signal`, returning the previous signal handler if it was
+/// `SigHandler::SigDfl` or `SigHandler::SigIgn`, `None` otherwise.
+///
+/// # Examples
+///
+/// ```no_run
+/// use nix::sys::signal::{Signal, SigHandler, signal};
+///
+/// unsafe {
+///     signal(Signal::SIGINT, SigHandler::SigIgn).unwrap();
+///     assert_eq!(signal(Signal::SIGINT, SigHandler::SigDfl).unwrap(), Some(SigHandler::SigIgn));
+/// }
+/// ```
+///
+/// # Errors
+/// Returns any error from `libc::signal`, such as when setting a signal handler for
+/// `SIGKILL` or `SIGSTOP`.
+///
+/// ```no_run
+/// use nix::sys::signal::{Signal, SigHandler, signal};
+///
+/// unsafe {
+///     assert!(signal(Signal::SIGKILL, SigHandler::SigIgn).is_err());
+///     assert!(signal(Signal::SIGSTOP, SigHandler::SigIgn).is_err());
+/// }
+/// ```
+pub unsafe fn signal(signal: Signal, handler: SigHandler) -> Result<Option<SigHandler>> {
+    let signal = signal as libc::c_int;
+    let res = match handler {
+        SigHandler::SigDfl => libc::signal(signal, libc::SIG_DFL),
+        SigHandler::SigIgn => libc::signal(signal, libc::SIG_IGN),
+        SigHandler::Handler(handler) => libc::signal(signal, handler as libc::sighandler_t),
+        SigHandler::SigAction(handler) => libc::signal(signal, handler as libc::sighandler_t),
```

Comment:
Is it valid to pass a `sa_sigaction` function pointer to `signal(3)`?  It may work on some OSes, but the man page doesn't say anything about it.  I think we should force people to use `sigaction(2)` if they want to do that.

Reply:
you're totally right, signal(3) specifies `sighandler_t`